benchmark/fft/fft_runner.h

```diff
@@ -0,0 +1,115 @@
+      : reporter_(reporter) {}
+
+  void SetInputs(const std::vector<PolyOrEvals>* polys,
+                 std::vector<std::unique_ptr<Domain>>&& domains) {
+    polys_ = polys;
+    domains_ = std::move(domains);
+  }
+
+  template <typename Fn, typename RetPoly>
+  void Run(Fn fn, const std::vector<uint64_t>& degrees,
+           std::vector<RetPoly>* results) {
+    for (size_t i = 0; i < degrees.size(); ++i) {
+      base::TimeTicks now = base::TimeTicks::Now();
+      if constexpr (std::is_same_v<PolyOrEvals, typename Domain::Evals>) {
+        std::unique_ptr<tachyon_bn254_univariate_dense_polynomial> ret;
+        ret.reset(fn(
+            reinterpret_cast<const tachyon_bn254_univariate_evaluation_domain*>(
+                domains_[i].get()),
+            reinterpret_cast<const tachyon_bn254_univariate_evaluations*>(
+                &polys_->at(i))));
+        reporter_->AddResult(i, (base::TimeTicks::Now() - now).InSecondsF());
+        results->push_back(*reinterpret_cast<RetPoly*>(ret.get()));
+        // clang-format off
+      } else if constexpr (std::is_same_v<PolyOrEvals, typename Domain::DensePoly>) {
+        // clang-format on
+        std::unique_ptr<tachyon_bn254_univariate_evaluations> ret;
+        ret.reset(fn(
+            reinterpret_cast<const tachyon_bn254_univariate_evaluation_domain*>(
+                domains_[i].get()),
+            reinterpret_cast<const tachyon_bn254_univariate_dense_polynomial*>(
+                &polys_->at(i))));
+        reporter_->AddResult(i, (base::TimeTicks::Now() - now).InSecondsF());
+        results->push_back(*reinterpret_cast<RetPoly*>(ret.get()));
+      }
+    }
+  }
+
+  template <typename RetPoly>
+  void RunExternal(FFTExternalFn fn, const std::vector<uint64_t>& exponents,
+                   std::vector<RetPoly>* results) const {
+    for (size_t i = 0; i < exponents.size(); ++i) {
+      uint64_t duration_in_us;
+      const F omega = domains_[i]->group_gen();
+      const F omega_inv = omega.Inverse();
+      if constexpr (std::is_same_v<PolyOrEvals, typename Domain::Evals>) {
+        std::unique_ptr<F> ret;
+        ret.reset(reinterpret_cast<F*>(
+            fn(reinterpret_cast<const tachyon_bn254_fr*>(
+                   polys_->at(i).evaluations().data()),
+               polys_->at(i).Degree(),
+               reinterpret_cast<const tachyon_bn254_fr*>(&omega_inv),
+               exponents[i], &duration_in_us)));
+        std::vector<F> res_vec(ret.get(), ret.get() + polys_->at(i).Degree());
+        results->emplace_back(
+            typename RetPoly::Coefficients(std::move(res_vec)));
+        // clang-format off
+      } else if constexpr (std::is_same_v<PolyOrEvals, typename Domain::DensePoly>) {
+        // clang-format on
+        std::unique_ptr<F> ret;
+        ret.reset(reinterpret_cast<F*>(
+            fn(reinterpret_cast<const tachyon_bn254_fr*>(
+                   polys_->at(i).coefficients().coefficients().data()),
```

Comment:
Why did you use `polys_->at(i)`? If you are sure with range, then please use `(*polys_)[i]` instead.